pkg/redirectpolicy/manager.go

```diff
@@ -459,6 +591,15 @@ func (rpm *Manager) deletePolicyFrontend(config *LRPConfig, frontend *frontend)
 		log.WithError(err).Debugf("Local redirect service for policy %v not deleted",
 			config.id)
 	}
+	if config.skipRedirectFromBackend {
+		// Delete skip_lb map entries.
+		addr := frontend.AddrCluster
+		if addr.Is4() {
+			rpm.skipLBMap.DeleteLB4ByAddrPort(addr.AsNetIP(), frontend.Port)
```

Comment:
We don't seem to handle BPF syscall errors from the skip map operations. At the very least we could use `WithCache()` and have a retry mechanism?

Reply:
I checked errors paths for other BPF maps (`git grep Map.Delete`) in the code base, and I'm not seeing any retries being done, only error being logged.

Reply:
Yeah that's a general endemic problem in the codebase. We're missing critical error handling in way too many places. Looks like you're using `pkg/ebpf` and not `pkg/bpf` so there's no `WithCache()`. I guess this is less critical feature so it can live a short while without proper resilience? Alternatively you could switch the `ebpf.NewMap` to using `bpf.NewMap` and then you can chain it with `WithCache()` and have a controller that retries failed operations.

I'm working on tooling to manage BPF maps with resilience (StateDB, reconciler and https://github.com/cilium/cilium/pull/32123). I'll probably do a pass to switch simple maps over to this in coming months and can also do it for this map.

Reply:
I agree that having a controller to retry failed operations could make the maps operation code more resilient in general. However, I would be curious to know what failure scenarios could really benefit from such retries.